File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/PreCheckoutQueryFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface PreCheckoutQueryFilter {

    UserFilter from() default @UserFilter(status = AnnotationState.OFF);

    StringFilter currency() default @StringFilter(status = AnnotationState.OFF);

    StringFilter totalAmount() default @StringFilter(status = AnnotationState.OFF);

    StringFilter invoicePayload() default @StringFilter(status = AnnotationState.OFF);

    StringFilter shippingOptionId() default @StringFilter(status = AnnotationState.OFF);

    OrderInfoFilter orderInfo() default @OrderInfoFilter(status = AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/composite/PreCheckoutQueryFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.composite

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class PreCheckoutQueryFilter(
        val from: UserFilter = UserFilter(status = AnnotationState.OFF),
        val validator: Array<String> = [],
        val currency: StringFilter = StringFilter(status = AnnotationState.OFF),
        val totalAmount: StringFilter = StringFilter(status = AnnotationState.OFF),
        val invoicePayload: StringFilter = StringFilter(status = AnnotationState.OFF),
        val shippingOptionId: StringFilter = StringFilter(status = AnnotationState.OFF),
        val orderInfo: OrderInfoFilter = OrderInfoFilter(status = AnnotationState.OFF)
)
